docs/develop-resources/api-conventions/validations.md

```diff
@@ -57,8 +57,18 @@ type CloudBuildWorkerPoolSpec struct {
 
 Validates the immutability in the controller runtime, by comparing the difference between `spec` and `status`, together with the corresponding proto field "IMMUTABLE" **field_behavior**. (default behavior if using the direct auto-generated code)
 
-To see if you can use this approach, check if the resource proto supports **field_behavior** (See [AIP 203](https://google.aip.dev/203)). 
-If so, you just need to make sure
+All google services are located in [googleapis Github repo](https://github.com/googleapis/googleapis/tree/master/google),
+refer to your resource's API documentation to identify the service name, for example: [compute](https://cloud.google.com/compute/docs/reference/rest/v1).
+Once you identify the service, find the proper path to the proto files, for example:
+[`google/cloud/compute/v1/compute.proto`](https://github.com/googleapis/googleapis/blob/master/google/cloud/compute/v1/compute.proto).
+To see if you can use this approach, check if the resource proto supports **field_behavior** (See [AIP 203](https://google.aip.dev/203)).
+
+Search for the resource name, like `message TargetTcpProxy`, this resource does not support immutable field_behavior.
+SecureSourceManagerInstance resource, for example `message Instance` in [secure_source_manager.proto](https://github.com/googleapis/googleapis/blob/master/google/cloud/securesourcemanager/v1/secure_source_manager.proto),
+field [`kms_key`](https://github.com/googleapis/googleapis/blob/master/google/cloud/securesourcemanager/v1/secure_source_manager.proto#L398)(KCC type KmsKeyRef) 
+has `(google.api.field_behavior) = IMMUTABLE`.
```

Comment:
However, I noticed that KmsKeyRef field and other immutable fields does not support this immutable check([code](https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/885f4af8544925fc4ad02dc3653b499eb08aad44/apis/securesourcemanager/v1alpha1/instance_types.go#L44-L45))... we'll probably need to update them later.

Similar to Compute types, will be addressed in b/377531379.